src/vsg/app/RecordTraversal.cpp

```diff
@@ -484,7 +485,14 @@ void RecordTraversal::apply(const CommandGraph& commandGraph)
     if (recordedCommandBuffers)
     {
         auto cg = const_cast<CommandGraph*>(&commandGraph);
-        cg->record(recordedCommandBuffers, _frameStamp, _databasePager);
+        if (cg->device)
+        {
+            cg->record(recordedCommandBuffers, _frameStamp, _databasePager);
+        }
+        else
+        {
+            warn("RecordTraversal::apply(const CommandGraph& commandGraph) cannot traverse as commandGraph->device = ", cg->device);
```

Comment:
cg->device would be always be nullptr here, so maybe simplify that warning by getting rid of the (always-nullptr) parameter and write that directly into the text string?